idea/vara-eth/frontend/src/components/ui/pagination/pagination.tsx

```diff
@@ -0,0 +1,44 @@
+import ArrowLeftSVG from '@/assets/icons/arrow-square-left.svg?react';
+import { Button } from '@/components';
+
+import styles from './pagination.module.scss';
+
+type PaginationProps = {
+  currentPage: number;
+  totalPages: number;
+  onPageChange: (page: number) => void;
+};
+
+const Pagination = ({ currentPage, totalPages, onPageChange }: PaginationProps) => {
+  if (!totalPages || totalPages <= 1) return null;
+
+  const handlePrev = () => {
+    if (currentPage > 1) {
+      onPageChange(currentPage - 1);
+    }
+  };
+
+  const handleNext = () => {
+    if (currentPage < totalPages) {
+      onPageChange(currentPage + 1);
+    }
+  };
```

Comment:
```suggestion
  const handlePrev = () => {
    onPageChange(prevPage => prevPage - 1);
  };

  const handleNext = () => {
    onPageChange(prevPage => prevPage + 1);
  };
```

Should be updated using updater function. Check guards aren't needed - it's handled by button state in UI.